src/apps/weblib/objectregistry.h

```diff
@@ -10,33 +12,43 @@ namespace Vizzu
 class ObjectRegistry
 {
 public:
-	using Handle = void *;
+	using Handle = const void *;
 
-	Handle reg(const std::shared_ptr<void> &ptr)
+	template <class T> Handle reg(std::shared_ptr<T> ptr)
 	{
-		Handle handle = ptr.get();
-		objects.emplace(handle, ptr);
-		return handle;
+		Handle res{ptr.get()};
+		{
+			auto lock = std::lock_guard{mutex};
```

Comment:
I think the "C-api" is like an API that can be used from multiple threads. I don't think to it that it is a JS API. 

But the emscripten supports the threading, so if someone wants to use it that way (workers) this will work.